ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/MLModelTool.java

```diff
@@ -54,6 +55,7 @@ public class MLModelTool implements Tool {
     private Parser inputParser;
     @Setter
     @Getter
+    @VisibleForTesting
```

Comment:
Why do we need this?

Reply:
Make UT be able to access the private field `outputParser` and test whether its results are as expected.

Reply:
But in the test you already declared: `Parser outputParser = new MLModelTool(client, "modelId", "response").getOutputParser();` How are you using `outputParser` from the MLModelTool class then?